Task50/UnitTest1.cs

```diff
@@ -0,0 +1,209 @@
+            {
+                selectedSates.SelectByIndex(rnd.Next(1, 8));
+            }
+
+            bool result = true;
+            for (int i = 0; i < 3; i++)
+            {
+                result = selectedSates.AllSelectedOptions[i].Selected;
+            }
+
+            Assert.IsTrue(result);
+        }
+
+        [Test]
+        public void ConfirmOkAlertTest()
+        {
+            _driver.Navigate().GoToUrl("https://www.seleniumeasy.com/test/javascript-alert-box-demo.html");
+            _driver.FindElement(By.XPath("//button[@onclick=\"myConfirmFunction()\"]")).Click();
+
+            try
+            {
+                IAlert confirmAlert = _driver.SwitchTo().Alert();
+                confirmAlert.Accept();
+
+                IWebElement resultMessage = _driver.FindElement(By.Id("confirm-demo"));
+                Assert.AreEqual(resultMessage.Text, "You pressed OK!");
+              
+            }
+            catch(NoAlertPresentException)
+            {
+                Assert.Fail("No Alert was displayed");
+            }
+            finally
+            {
+                _driver.Close();
+            }
+            
+        }
+
+
+        [Test]
+        public void ConfirmCancelAlertTest()
+        {
+            _driver.Navigate().GoToUrl("https://www.seleniumeasy.com/test/javascript-alert-box-demo.html");
+            _driver.FindElement(By.XPath("//button[@onclick=\"myConfirmFunction()\"]")).Click();
+
+            try
+            {
+                IAlert confirmAlert = _driver.SwitchTo().Alert();
+                confirmAlert.Dismiss();
+
+                IWebElement resultMessage = _driver.FindElement(By.Id("confirm-demo"));
+                Assert.AreEqual(resultMessage.Text, "You pressed Cancel!");
+
+            }
+            catch (NoAlertPresentException)
+            {
+                Assert.Fail("No Alert was displayed");
+            }
+            finally
+            {
+                _driver.Close();
+            }
+
+        }
+
+        [TestCase("Test Name", "Test Name")]
+        [Test]
+        public void AlertBoxTest(string inValue, string outValue)
+        {
+            _driver.Navigate().GoToUrl("https://www.seleniumeasy.com/test/javascript-alert-box-demo.html");
+            _driver.FindElement(By.XPath("//button[@onclick=\"myPromptFunction()\"]")).Click();
+
+            try
+            {
+                IAlert confirmAlert = _driver.SwitchTo().Alert();
+                confirmAlert.SendKeys(inValue);
+                confirmAlert.Accept();
+
+                IWebElement resultMessage = _driver.FindElement(By.Id("prompt-demo"));
+                Assert.AreEqual(resultMessage.Text, $"You have entered '{outValue}' !", "Actual value is not equal to the entered text");
+
+            }
+            catch (NoAlertPresentException)
+            {
+                Assert.Fail("No Alert was displayed");
+            }
+            finally
+            {
+                _driver.Close();
```

Comment:
This line have to put in TearDown method with [TearDown] attribute. It will allow to close driver after each test running without code dublication.